cpp/benchmarks/synchronization/synchronization.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "synchronization.hpp"
+
+cuda_event_timer::cuda_event_timer::cuda_event_timer(benchmark::State& state, cudaStream_t stream_): p_state(&state), stream(stream_) {
```

Comment:
I think there's one too many `cuda_event_timer` here. 
```suggestion
cuda_event_timer::cuda_event_timer(benchmark::State& state, cudaStream_t stream_): p_state(&state), stream(stream_) {
```

Reply:
I am actually surprised that it compiled ...

Reply:
I'm reminded that it's a little known C++ eccentricity that you can add as many redundant scope prefixes as you want, e.g., https://wandbox.org/permlink/UTSk2060Sx9693gP